CONTRIBUTING.md

```diff
@@ -0,0 +1,109 @@
+* [pint](https://cloudflare.github.io/pint/checks/promql/series.html) - PromQL linting
+* [gitlint](https://jorisroovers.com/gitlint/) - commit message linting
+* [yamllint](https://yamllint.readthedocs.io/en/stable/) - YAML linting
+
+### Unit Tests
+PromQL unit tests are stored in YAML files under [test/promql/tests](test/promql/tests).
+
+Prometheus rules, stored as Kubernetes resources, are defined
+[here](prometheus/base/prometheus.rules.yaml).
+The Prometheus-related tools used for unit tests (and linting) cannot digest this
+directly, so instead, the `prepare` target on the [Makefile](Makefile) extracts the
+Prometheus configurations out of those configurations and stores them to a temporary
+location: `test/promql/extracted-rules.yaml`.
+
+The different test procedures should then refer to that temporary location as the 
+location of the Prometheus rules. For example, a `promtool` test file
+`test/promql/tests/my-test.yaml` should point to the rule file as follows:
+
+```
+rule_files:
+  - ../extracted-rules.yaml
+```
+
+## Pull Requests
+This section covers a few aspect to have in mind while writing/reviewing Pull Requests
+
+### Test Coverage
+Each Pull Request (PR) should include full test coverage for the code it introduces.
+
+### Pull Request Span
+Pull requests should be as small in size as possible.
+
+The commits within a PR should convey the progression of the code between the point
+before the PR was merged and after it was merged. They should NOT convey the progression
+of the code within the PR lifecycle (e.g. no `addressing comments` commit messages).
+
+### Commit Messages
+The project is using [gitlint](https://jorisroovers.com/gitlint/) for enforcing commit
+message structure. The linting rules are defined [here](.gitlint) and are enforced by
+the CI system.
+
+[Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) is used as
+the commit message standard.
+
+If the commit message (also applicable to PR names and branch names) includes a name of
+a Jira ticket, then it will automatically be referenced within the Jira ticket.
+
+commit messages should be detailed:
+* The header should describe the purpose of the commit.
+* The body should be written as if the reviewer does not know the story behind the
+  commit.
+* The body should provide the motivation behind the commit and an overview of how it
+  works.
+
+Example:
+```
+chore(STONEO11Y-21): add network egress metric and tests
+
+- Create a new rule for getting the network transmit in bytes and
+and add the label_pipelines_appstudio_openshift_io_type to it.
+
+- Add a Unit test file to test the metric and its products.
+
+Signed-off-by: Avi Biton <abiton@redhat.com>
+```
+
+### Code Review Guidelines
+* Each PR should be approved by at least 2 team members. Those approvals are only
+relevant if given since the last major change in the PR content.
+
+* All comments raised during code review should be addressed (fixed/replied).
```

Comment:
Maybe we should have a guideline as to resolving comments (Reviewer/PR assignee?), this issue always comes up.

Reply:
I've added something. Let me know your thoughts.

Reply:
I like it!